docs/source/guides/build.rst

```diff
@@ -0,0 +1,218 @@
+
+.. hint:: Please make sure that Docker is running before executing the script
+
+After you execute this script, following things happen:
+
+1. The script checks if you don't have containers with Iroha already running.
+It ends up with reattaching you to interactive shell upon successful completion.
+2. The script will download ``iroha-docker-develop`` and ``postgres`` images. 
+``iroha-docker-develop`` image contains all development dependencies, and is 
+based on top of ``ubuntu:16.04``. ``postgres`` image is required for starting 
+and running Iroha.
+3. Two containers are created and launched.
+4. The user is attached to the interactive environment for development and 
+testing with ``iroha`` folder mounted from the host machine. Iroha folder 
+is mounted to ``/opt/iroha`` in Docker container.
+
+Now your are ready to build Iroha! Please go to `Building Iroha` section.
+
+.. note::  Docker environment will be removed when you detach from the
+  container.
+
+Linux
+^^^^^
+
+Boost
+"""""
+
+Iroha requires Boost of at least 1.65 version.
+To install Boost libraries (``libboost-all-dev``), use `current release 
+<http://www.boost.org/users/download/>`_ from Boost webpage. The only 
+dependencies are system and filesystem, so use 
+``./bootstrap.sh --with-libraries=system,filesystem`` when you are building 
+the project.
+
+Other dependencies
+""""""""""""""""""
+
+To build Iroha, you need following packages:
+
+``build-essential`` ``automake`` ``libtool`` ``libssl-dev`` ``zlib1g-dev`` 
+``libc6-dbg`` ``golang`` ``git`` ``ssh`` ``tar`` ``gzip`` ``ca-certificates``
+``wget`` ``curl`` ``file`` ``unzip`` ``python`` ``cmake``
+
+Use this code to install dependencies on Debian-based Linux distro.
+
+.. code-block:: shell
+
+  apt-get update; \
+  apt-get -y --no-install-recommends install \
+  build-essential automake libtool \
+  libssl-dev zlib1g-dev \
+  libc6-dbg golang \
+  git ssh tar gzip ca-certificates \
+  wget curl file unzip \
+  python cmake
+
+.. note::  If you are willing to actively develop Iroha, please consider installing 
+  the `latest release <https://cmake.org/download/>`_ of CMake.
+
+macOS
+^^^^^
+
+If you want to build it from scratch and actively develop it, please use this code 
+to install all dependencies with Homebrew.
+
+.. code-block:: shell
+
+  xcode-select --install
+  brew install cmake boost postgres grpc autoconf automake libtool golang libpqxx
+
+.. hint:: To install the Homebrew itself please run 
+
+  ``ruby -e "$(curl -fsSL https://raw.githubusercontent.com/homebrew/install/master/install)"``
+
+Build Process
+-------------
+
+Cloning the Repository
+^^^^^^^^^^^^^^^^^^^^^^
+Clone the `Iroha repository <https://github.com/hyperledger/iroha>`_ to the
+directory of your choice.
+
+.. code-block:: shell
+
+  git clone -b develop https://github.com/hyperledger/iroha
+  cd iroha
+
+.. hint:: If you have installed the prerequisites with Docker, you don't need
+  to clone Iroha again
```

Comment:
…because when you run `run-iroha-dev.sh` it attaches to Iroha source code folder. Feel free to edit source code files with your host environment and build it within docker container.